docs/ref/models/fields.txt

```diff
@@ -103,6 +115,12 @@ and the second element is the human-readable name. For example::
         ("GR", "Graduate"),
     ]
 
+.. versionchanged: 5.0
+
+    Dictionary support was added. Previously Django only supported lists of
+    two-tuples, but since Python 3.6+ dictionaries preserve insertion order
```

Comment:
Would 2-tuple be more mathematically correct? (honest question!)

Reply:
I'm not sure maths really comes into this, but yeah, why not! 😄

Reply:
I had a general look and we seem to largely prefer `<number-digit>-tuple` to `<number-word>-tuple`.

There are a bunch of existing cases to update:

```console
$ git grep --extended --line-number '(one|two|three|four|five|six)-tuple'
django/contrib/admindocs/views.py:459:    Each object in the returned list is a four-tuple:
django/db/backends/base/schema.py:1349:        Return a two-tuple of: an SQL fragment of (sql, params) to insert into
docs/ref/contrib/admin/index.txt:345:    ``fieldsets`` is a list of two-tuples, in which each two-tuple represents a
docs/ref/contrib/admin/index.txt:349:    The two-tuples are in the format ``(name, field_options)``, where ``name``
docs/ref/contrib/admin/index.txt:1559:    of two-tuples, in which each two-tuple represents a ``<fieldset>`` on the
docs/ref/contrib/gis/geoquerysets.txt:896:Returns the extent of all ``geo_field`` in the ``QuerySet`` as a four-tuple,
docs/ref/contrib/gis/geoquerysets.txt:918:Returns the 3D extent of all ``geo_field`` in the ``QuerySet`` as a six-tuple,
docs/ref/models/fields.txt:261:  value will be a ``(value, label)`` two-tuple. See below for :ref:`an example
docs/ref/models/querysets.txt:768:rather than one-tuples. An example should make the difference clearer:
docs/ref/settings.txt:1962:The list is a list of two-tuples in the format
docs/ref/signals.txt:461:    necessary to know the plan. A plan is a list of two-tuples with the first
docs/ref/signals.txt:519:    know the plan. A plan is a list of two-tuples with the first item being
docs/releases/1.11.1.txt:29:* Fixed a crash when using a two-tuple in ``EmailMessage``’s ``attachments``
```

I'll open a separate PR for those and have updated the new changes in this PR to reflect that style.

Reply:
See https://github.com/django/django/pull/16996 for fixing of all existing cases.